packages/orchestrator-ui-components/src/api/index.ts

```diff
@@ -0,0 +1,132 @@
+        showErrorDialog = true,
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        result = true,
+    ): Promise<R> => {
+        // preset the config with the relative URL and a GET type.
+        // presets can be overridden with `options`.
+        return axiosInstance({ url: path, method: 'GET', ...options })
+            .then((res) => res.data)
+            .catch((err) => {
+                if (showErrorDialog) {
+                    console.debug(
+                        `Show error dialog for GET ${path} returning ${err?.response?.status}`,
+                    );
+                    setTimeout(() => {
+                        throw err;
+                    }, 250);
+                }
+                throw err;
+            });
+    };
+
+    catchErrorStatus = <T>(
+        promise: Promise<unknown>,
+        status: number,
+        callback: (json: T) => void,
+    ) => {
+        return promise.catch((err) => {
+            if (err.response && err.response.status === status) {
+                callback(err.response.data);
+            } else {
+                throw err;
+            }
+        });
+    };
+
+    fetchJson = <R = object>(
+        path: string,
+        options = {},
+        headers = {},
+        showErrorDialog = true,
+        result = true,
+    ): Promise<R> => {
+        return this.axiosFetch(path, options, headers, showErrorDialog, result);
+    };
+
+    fetchJsonWithCustomErrorHandling = <R = object>(
+        path: string,
+    ): Promise<R> => {
+        return this.fetchJson(path, {}, {}, false, true);
+    };
+
+    postPutJson = <R = object>(
+        path: string,
+        processInput: object,
+        method: string,
+        showErrorDialog = true,
+        result = true,
+    ): Promise<R> => {
+        return this.axiosFetch(
+            path,
+            { method: method, data: processInput },
+            {},
+            showErrorDialog,
+            result,
+        );
+    };
+}
+
+abstract class ApiClientInterface extends BaseApiClient {
+    abstract cimStartForm: (
+        formKey: string,
+        userInputs: object[],
+    ) => Promise<object>;
+}
+
+class ApiClient extends ApiClientInterface {
+    startProcess = (
+        workflow_name: string,
+        processInput: object,
+    ): Promise<unknown> => {
+        return this.postPutJson<unknown>(
+            'processes/' + workflow_name,
+            processInput,
+            'post',
+            false,
+            true,
+        );
+    };
+    products = (): Promise<ProductDefinition[]> => {
+        return this.fetchJson<ProductDefinition[]>('products/');
+    };
+    productById = (productId: string): Promise<ProductDefinition> => {
+        return this.fetchJson(`products/${productId}`);
+    };
+    cimStartForm = (
+        formKey: string,
+        userInputs: object[],
+    ): Promise<{ id: string }> => {
+        return this.postPutJson(
+            `surf/forms/${formKey}`,
```

Comment:
There are multiple urls or partial URLs here. Maybe move them to the top of the file as a const of some other convenient place.

Reply:
Done